rfcs/2021-06-15-shipping-blade.md

```diff
@@ -0,0 +1,438 @@
+      "default": "./build/tokens/index.web.js"
+    }
+  }
+}
+```
+
+With the above approach we tell tools like webpack, metro, node on how to resolve subpaths. Now after above addition we should be able to do like following:
+```js
+import {ThemeProvider, useTheme} from '@razorpay/blade/components';
+import {paymentTheme, globalColors} from '@razorpay/blade/tokens';
+```
+The downside of `exports` map is that not all the bundler tools have implemented it. Webpack has [it in v5+](https://github.com/webpack/webpack/pull/10953) and metro [hasn't implemented](https://github.com/facebook/metro/issues/670) this yet.
+
+> You can read more about the [exports map here](https://docs.skypack.dev/package-authors/package-checks#export-map). I've added few more links in the [references section](#references) below.
+
+### `main`, `module`, `browser`, `react-native` fields in `package.json`
+
+The older way to define this consumption information used to be via `main`, `module`, `browser`, `react-native` fields in `package.json`.
+* `main`: defines entry point for commonjs bundle
+* `module`: defines entry point for esm bundle.
+* `browser`: defines entry point for browser specific bundle. If not provided, then fallback to `main` or `module`.
+* `react-native`: defines entry point for react-native bundle. If not provided, then fallback to `main`.
+* The problem with all of the above fields is that they accept a single string value and can be useful if you have single bundle of your package. For our use case we have multiple bundles. Hence, this won't be useful for us.
+
+### Conclusion
+Now based on the above discussions we definitely want to go ahead with `exports` map to define subpaths for our package but that is not supported by all the tools. So what do we do? We need to provide fallback for backward compatibility. So how do we do that? We generate `components`, `tokens`, `utils`  files at the root and re-export things from `build`.
+```
+blade/
+  build/
+    components/
+      index.web.js
+      index.native.js
+    tokens/
+      index.web.js
+      index.native.js
+    utils/
+      index.web.js
+      index.native.js
+  src/
+  components.js
+    export * from './build/components';
+  tokens.js
+    export * from './build/tokens';
+  utils.js
+    export * from './build/utils';
+```
+
+With the above we would still be able to import in following way:
+```js
+import {ThemeProvider, useTheme} from '@razorpay/blade/components';
+import {paymentTheme, globalColors} from '@razorpay/blade/tokens';
+```
+
+# Drawbacks/Constraints
+I've already spoken about multiple approaches and constraints of each of them in the respective sections itself.
+
+# Alternatives
+I've already spoken about multiple approaches and constraints of each of them in the respective sections itself.
+
+# Adoption strategy
+There's no adoption strategy as such. We'll just need to document the usage of design system once it's published.
```

Comment:
Can just add a line saying that thinking of easier & smoother adoption, we've taken [x] approach.